oga/backend/users/views.py

```diff
@@ -1,42 +1,68 @@
 from django.shortcuts import render, get_object_or_404
-from django.http import HttpResponse, HttpResponseBadRequest, JsonResponse
+from django.http import HttpResponse, HttpResponseBadRequest, JsonResponse, HttpResponseNotAllowed
 from django.views.decorators.csrf import csrf_exempt
 import json
 from .models import Profile, Question, Answer, Location
 from django.contrib.auth.models import User
 from django.contrib.auth import login, authenticate, get_user
 from django.views import generic
+from functools import wraps
 
+def login_required(views_func):
+    @wraps(views_func)
+    def wrapper(request, *args, **kwargs):
+        if request.user.is_authenticated:
+            return views_func(request, *args, **kwargs)
+        else:
+            return HttpResponse(status=400)
+    return wrapper
+
+def check_request(views_func):
+    @wraps(views_func)
+    def wrapper(request, *args, **kwargs):
+        try:
+            return views_func
+        except (KeyError, ValueError) as e:
+            return HttpResponse(status=400)
+    return wrapper
 
-def index(request):
-    return HttpResponse('Hello World!')
+# def index(request):
+#     return HttpResponse('Hello World!')
 
+@login_required
+@check_request
 def UserProfile(request, username):
+    if not (request.user.username == username):
```

Comment:
Django has something along the lines of check_permission decorator. Maybe we can use them instead.

Also, I think dividing user profile view to two (one for self, and one for viewing others perhaps?)
might be a better idea, design choice wise. Since this commit is not directly related to error_handling, I think we should discuss our design choice before implementing this part.

Reply:
You mean from django.contrib.auth.decorators import permission_required?
As you mentioned I searched it in the internet but it looks like it deals with what kind of things can be done regarding specific kind of user group.
In this case, we just need to find if there username matches or not.
If I didn't get your words well, then could you fix me?

Reply:
Thank you for the information. I probably should search more before leaving this kind of comments :(. I think you are right.